axelrod/strategies/axelrod_second.py

```diff
@@ -975,3 +975,297 @@ def strategy(self, opponent: Player) -> Action:
                 self.defect_padding = 0
 
             return self.try_return(opponent.history[-1])
+
+
+class Harrington(Player):
+    """
+    Strategy submitted to Axelrod's second tournament by Paul Harrington (K75R)
+    and came in eighth in that tournament.
+
+    This strategy has three modes:  Normal, Fair-weather, and Defect.  These
+    mode names were not present in Harrington's submission.
+
+    In Normal and Fair-weather modes, the strategy begins by:
+
+    - Update history
+    - Detects random if turn is multiple of 15 and >=30.
```

Comment:
randomly

Reply:
Detects random opponent? (Not overly fussed)